File: src/main/java/me/miki/shindo/utils/vector/Vertex.java
package me.miki.shindo.utils.vector;

import lombok.Getter;
import org.lwjgl.util.vector.Vector3f;

@Getter
public class Vertex {

    private final Vector3f pos;
    private final float u;
    private final float v;

    public Vertex(float f, float g, float h, float i, float j) {
        this(new Vector3f(f, g, h), i, j);
    }

    public Vertex(Vector3f vector3f, float f, float g) {
        this.pos = vector3f;
        this.u = f;
        this.v = g;
    }

    public Vertex remap(float f, float g) {
        return new Vertex(this.pos, f, g);
    }

}

File: src/main/kotlin/me/miki/shindo/utils/vector/Vertex.kt
package me.miki.shindo.utils.vector

/**
 * Lightweight vertex container used by the 3D skin renderer.
 */
class Vertex(val pos: Vector3f, val u: Float, val v: Float) {

    constructor(x: Float, y: Float, z: Float, u: Float, v: Float) : this(Vector3f(x, y, z), u, v)

    /**
     * Returns a copy with the same position and new UVs.
     */
    fun setTexturePosition(u: Float, v: Float): Vertex {
        return Vertex(Vector3f(pos.x, pos.y, pos.z), u, v)
    }
}
